src/utilities/common.ts

```diff
@@ -213,11 +223,16 @@ export function modelFromFieldConfigs (fieldConfigs: IFieldConfigPartial[], data
         const { field, nullify } = fieldConfig
           , formValue = get(data, field)
           , shouldNullify = nullify && !formValue && formValue !== false
-          , value = shouldNullify ? null : formValue
-          ;
+          , value = isTypeObjectSearchCreate(fieldConfig) && isObject(formValue) && !has(formValue, ID_ATTR)
```

Comment:
Could you split this logic into another boolean so the name can provide some context for this? For example, I don't understand why `!has(formValue, ID_ATTR)` is important on whether to recurse

Reply:
I tried my best. I agree it's kinda awkward. Let me know if this helps.